openstack/blockstorage/v1/snapshots/update_metadata.go

```diff
@@ -0,0 +1,30 @@
+package snapshots
+
+import (
+	"github.com/opentelekomcloud/gophertelekomcloud"
+	"github.com/opentelekomcloud/gophertelekomcloud/internal/extract"
+)
+
+type UpdateMetadataOpts struct {
+	Metadata map[string]interface{} `json:"metadata,omitempty"`
+}
+
+func UpdateMetadata(client *golangsdk.ServiceClient, id string, opts UpdateMetadataOpts) (*map[string]interface{}, error) {
```

Comment:
more detail pls

Reply:
`UpdateMetadataOpts` contains only one field, so it can be simplified to passing simple `metadata map[string]interface{}` argument (though I'm not sure it will simplify the code)